### ISSUE TITLE
Concatenate Index Files fails for some inputs

### ISSUE THREAD

**Issue by atruskie** [QutEcoacoustics/audio-analysis#170]
### Expected behaviour

ConcatenateIndexFiles should work.

### Actual behaviour

A `System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.` exception is thrown.

### Steps to reproduce the behaviour

- What command did you use?
  - `"F:\FalseColor\Run-ParallelJobs\131738377986629221\0\AP\AnalysisPrograms.exe" concatenateindexfiles "Y:\Results\20180608-103353 - Tshering, Towsey.Indices, #216\Tshering\WBH_Walaytar\WBH_2018\Bermo" -o "Y:\Results\20180608-103353 - Tshering, Towsey.Indices, #216\ConcatResult\TimedGaps\Tshering\WBH_Walaytar\WBH_2018" --directory-filter *.wav -ip "Y:\Results\20180608-103353 - Tshering, Towsey.Indices, #216\IndexPropertiesConfig.yml" -fcs "Y:\Results\20180608-103353 - Tshering, Towsey.Indices, #216\SpectrogramFalseColourConfig.yml" -f Bermo -z 06:00 --gap-rendering TimedGaps --draw-images true -n`
- What config file did you use?
   - The standard config files
- Where can we find the source data used?
  - `Y:\Results\20180608-103353 - Tshering, Towsey.Indices, #216\Tshering\WBH_Walaytar\WBH_2018\Bermo`
- Program version?
  - `QUT Ecoacoustics Analysis Programs - version 18.6.0.34 (RELEASE build, 2018-06-10T14:02:04.2309201Z)
Git branch-version: master-2cadfe7f0e67684751371a1deaf2224fd0ed6c78, DirtyBuild:False, CI:447`

### Any other details

It appears to crash on the second day.

Log: Y:\Results\20180608-103353 - Tshering, Towsey.Indices, #216\ConcatResult\TimedGaps\Tshering\WBH_Walaytar\WBH_2018\Bermo\log.txt

**Comment by atruskie** [comment]
This bug is repeatable in other datasets:

`Y:\Results\20180802-125012 - Hongxiao, Towsey.Indices, ICD=15.0, #225\ConcatResult\TimedGaps\Hongxiao\KiwiRecordings\log_20180807T040638Z.txt`

```
"F:\FalseColor\Run-ParallelJobs\131780883729240605\3\AP\AnalysisPrograms.exe" concatenateindexfiles "Y:\Results\20180802-125012 - Hongxiao, Towsey.Indices, ICD=15.0, #225\Hongxiao\KiwiRecordings" -o "Y:\Results\20180802-125012 - Hongxiao, Towsey.Indices, ICD=15.0, #225\ConcatResult\TimedGaps\Hongxiao" --directory-filter *.wav -ip "Y:\Results\20180802-125012 - Hongxiao, Towsey.Indices, ICD=15.0, #225\IndexPropertiesConfig.yml" -fcs "Y:\Results\20180802-125012 - Hongxiao, Towsey.Indices, ICD=15.0, #225\SpectrogramFalseColourConfig.yml" -f KiwiRecordings -z 10:00 --gap-rendering TimedGaps --draw-images -n
```

```
QUT Ecoacoustics Analysis Programs - version 18.7.1.11 (RELEASE build, 2018-07-29T14:02:47.4284203Z)
Git branch-version: master-d4654f21196a1892d3737edb04e7f20351c2bec9, DirtyBuild:False, CI:50
```

**Comment by atruskie** [comment]
I've replicated this bug again: 
[log_20181018T032015Z.txt](https://github.com/QutEcoacoustics/audio-analysis/files/2490255/log_20181018T032015Z.txt)

**Comment by atruskie** [comment]
Closed by ce80cbcd019613db03446ba1189117ab42457afe and ce80cbcd019613db03446ba1189117ab42457afe
